aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt

```diff
@@ -1,35 +1,50 @@
 package org.wordpress.aztec.spans
 
 import android.content.Context
+import android.graphics.Bitmap
 import android.graphics.Canvas
 import android.graphics.Paint
 import android.graphics.Rect
+import android.graphics.drawable.BitmapDrawable
 import android.graphics.drawable.Drawable
 import android.view.Gravity
 import org.wordpress.aztec.AztecAttributes
 import org.wordpress.aztec.AztecText
 import java.util.*
 
-abstract class AztecMediaSpan(context: Context, drawable: Drawable?, override var attributes: AztecAttributes = AztecAttributes(),
+
+abstract class AztecMediaSpan(context: Context, imageProvider: IImageProvider, override var attributes: AztecAttributes = AztecAttributes(),
                               var onMediaDeletedListener: AztecText.OnMediaDeletedListener? = null,
-                              editor: AztecText? = null) : AztecDynamicImageSpan(context, drawable), IAztecAttributedSpan {
+                              editor: AztecText? = null) : AztecDynamicImageSpan(context, imageProvider), IAztecAttributedSpan {
 
     abstract val TAG: String
 
     private val overlays: ArrayList<Pair<Drawable?, Int>> = ArrayList()
 
+    private var innerPlaceholder: Bitmap? = null
+    private var isDirty = false
+    private val EXTRA_LOADING_SIZE = 500
+
     init {
         textView = editor
     }
 
-    fun setDrawable(newDrawable: Drawable?) {
-        imageDrawable = newDrawable
+    private fun drawableToBitmap(drawable: Drawable?): Bitmap? {
+        if (drawable == null || drawable !is BitmapDrawable || drawable.bitmap == null) {
+            return null
+        }
 
-        originalBounds = Rect(imageDrawable?.bounds ?: Rect(0, 0, 0, 0))
+        return Bitmap.createBitmap(drawable.bitmap.width, drawable.bitmap.height, Bitmap.Config.ALPHA_8)
+    }
 
-        setInitBounds(newDrawable)
+    fun setDrawable(newDrawable: Drawable?, isPlaceholder: Boolean = false) {
+        if (!isPlaceholder) {
+            innerPlaceholder = drawableToBitmap(newDrawable)
+        }
 
-        computeAspectRatio()
+        if (!isPlaceholder || innerPlaceholder == null) {
```

Comment:
@0nko I'm not sure the ` innerPlaceholder == null` is required here. Maybe it's not used in our projects, but we can change/switch the drawable of a `MediaSpan` resulting in an invalid `innerPlaceholder` since it was computed on the previous picture.

Reply:
ok :)